recipes/corrade/all/conanfile.py

```diff
@@ -90,8 +95,11 @@ def package(self):
         cmake.install()
 
         share_cmake = os.path.join(self.package_folder, "share", "cmake", "Corrade")
-        self.copy("UseCorrade.cmake", src=share_cmake, dst=os.path.join(self.package_folder, "lib", "cmake", "Corrade"))
+        self.copy("CMakeLists.txt", src=share_cmake, dst=os.path.join(self.package_folder, "lib", "cmake", "Corrade"))
+        self.copy("CorradeConfig.cmake", src=share_cmake, dst=os.path.join(self.package_folder, "lib", "cmake", "Corrade"))
         self.copy("CorradeLibSuffix.cmake", src=share_cmake, dst=os.path.join(self.package_folder, "lib", "cmake", "Corrade"))
+        self.copy("FindCorrade.cmake", src=share_cmake, dst=os.path.join(self.package_folder, "lib", "cmake", "Corrade"))
```

Comment:
What specific macro do you need from Corrade? Conan CmakeFindPackage generator should be enough. If there is no specific macro there, please remove FindCorrade/CorradeConfig

Reply:
All the different options passed to CMake when building corrade (that are relevant for consumers) gets written to a `configure.h` which is packaged along all other includes. This file [is then parsed by `FindCorrade.cmake`](https://github.com/mosra/corrade/blob/162d6a7d77e4940f7c05613013e1d3aa522bc8c6/modules/FindCorrade.cmake#L276-L318). The alternative is to extract the necessary pieces and add to a CMake script that's part of the recipe (if I'm correct it's only the parsing of `configure.h` that's relevant for consumers).

Or alternatively mimic it by directly in `package_info()` setup all these different preprocessor directives. Though this is even more error prone than above, IMO.

Reply:
I can't find a real evidence of their usage as cmake definitions. For instance, BUILD_STATIC will be "renamed" to CORRADE_BUILD_STATIC, and only the unit tests are using it. The defines in configure.h are okay, they will be processed by the consumer. Do you see a real usage for them?

https://github.com/mosra/corrade/search?q=CORRADE_UTILITY_USE_ANSI_COLORS
https://github.com/mosra/corrade/search?q=CORRADE_BUILD_STATIC